File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    answer = ''
    for i in range(1, len(food)):
        cnt = food[i] // 2
        
        answer += (str(i) * cnt)
        
    answer += '0'
    
    for i in range(len(food)-1, 0, -1):
        cnt = food[i] // 2
        
        answer += (str(i) * cnt)
    
    # print(answer)
        
    
    
    return answer

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

string solution(vector<int> food) {
    string answer = "";
    // answer += "123";
    
    // int a = 3;
    // cout << answer << endl;
    
    for (int i=1; i < food.size(); i++){
        int cnt = food[i] / 2;
        // cout << cnt << endl;
        
        for (int j = 0; j < cnt; j ++){
            answer += to_string(i);
        }
    }
    answer += "0";
    
    for (int i=food.size()-1; i > 0; i--){
        int cnt = food[i] / 2;
        // cout << cnt << endl;
        
        for (int j = 0; j < cnt; j ++){
            answer += to_string(i);
        }
    }    
    
    
    return answer;
}